src/ol_infrastructure/infrastructure/vector_log_proxy/__main__.py

```diff
@@ -0,0 +1,345 @@
+            description=f"Allow traffic to the vector-log-proxy server on port {VECTOR_LOG_PROXY_PORT}",
+        ),
+    ],
+    egress=default_egress_args,
+    vpc_id=ops_vpc_id,
+)
+
+###################################
+#     Web Node EC2 Deployment     #
+###################################
+
+# Create load balancer for Concourse web nodes
+vector_log_proxy_lb = lb.LoadBalancer(
+    "vector-log-proxy-load-balancer",
+    name=f"vector-log-proxy-alb-{stack_info.env_suffix}",
+    ip_address_type="dualstack",
+    load_balancer_type="application",
+    enable_http2=True,
+    subnets=target_vpc["subnet_ids"],
+    security_groups=[
+        vector_log_proxy_security_group.id,
+    ],
+    tags=aws_config.merged_tags({"Name": vector_log_proxy_tag}),
+)
+
+TARGET_GROUP_NAME_MAX_LENGTH = 32
+vector_log_proxy_lb_target_group = lb.TargetGroup(
+    "vector-log-proxy-alb-target-group",
+    vpc_id=ops_vpc_id,
+    target_type="instance",
+    port=VECTOR_LOG_PROXY_PORT,
+    protocol="HTTPS",
+    health_check=lb.TargetGroupHealthCheckArgs(
+        healthy_threshold=2,
+        interval=120,
+        path="/events",
+        port=str(VECTOR_LOG_PROXY_PORT),
+        protocol="HTTPS",
+        matcher="405",
+    ),
+    name=vector_log_proxy_tag[:TARGET_GROUP_NAME_MAX_LENGTH],
+    tags=aws_config.tags,
+)
+vector_log_proxy_acm_cert = acm.get_certificate(
+    domain="*.odl.mit.edu", most_recent=True, statuses=["ISSUED"]
+)
+vector_log_proxy_alb_listener = lb.Listener(
+    "vector-log-proxy-alb-listener",
+    certificate_arn=vector_log_proxy_acm_cert.arn,
+    load_balancer_arn=vector_log_proxy_lb.arn,
+    port=VECTOR_LOG_PROXY_PORT,
+    protocol="HTTPS",
+    default_actions=[
+        lb.ListenerDefaultActionArgs(
+            type="forward",
+            target_group_arn=vector_log_proxy_lb_target_group.arn,
+        )
+    ],
+)
+
+## Create auto scale group and launch configs for vector-log-proxy
+instance_type = (
+    vector_log_proxy_config.get("instance_type") or InstanceTypes.burstable_small.name
+)
+consul_datacenter = consul_stack.require_output("datacenter")
+
+grafana_credentials = read_yaml_secrets(
+    Path(f"vector/grafana.{stack_info.env_suffix}.yaml")
+)
+heroku_proxy_credentials = read_yaml_secrets(
+    Path(f"vector/heroku_proxy.{stack_info.env_suffix}.yaml")
+)
+
+vector_log_proxy_launch_config = ec2.LaunchTemplate(
+    "vector-log-proxy-launch-template",
+    name_prefix=f"vector-log-proxy-{stack_info.env_suffix}-",
+    description="Launch template for deploying vector-log-proxy nodes",
+    iam_instance_profile=ec2.LaunchTemplateIamInstanceProfileArgs(
+        arn=vector_log_proxy_instance_profile.arn,
+    ),
+    image_id=vector_log_proxy_ami.id,
+    block_device_mappings=[
+        ec2.LaunchTemplateBlockDeviceMappingArgs(
+            device_name="/dev/xvda",
+            ebs=ec2.LaunchTemplateBlockDeviceMappingEbsArgs(
+                volume_size=vector_log_proxy_config.get_int("disk_size") or 8,
```

Comment:
Probably worth defaulting to ~25GB since 8 can pretty quickly get overwhelmed, particularly if we start spooling any logs to disk while batching up groups.